homeassistant/components/google_assistant/trait.py

```diff
@@ -1076,22 +1079,33 @@ def query_attributes(self):
 
         if domain == cover.DOMAIN:
             position = self.state.attributes.get(cover.ATTR_CURRENT_POSITION)
-            if position is not None:
+            if params['openPercent'] == 0:
                 await self.hass.services.async_call(
-                    cover.DOMAIN, cover.SERVICE_SET_COVER_POSITION, {
-                        ATTR_ENTITY_ID: self.state.entity_id,
-                        cover.ATTR_POSITION: params['openPercent']
+                    cover.DOMAIN, cover.SERVICE_CLOSE_COVER, {
+                        ATTR_ENTITY_ID: self.state.entity_id
+                    }, blocking=True, context=data.context)
+            elif params['openPercent'] == 100:
+                await self.hass.services.async_call(
+                    cover.DOMAIN, cover.SERVICE_OPEN_COVER, {
+                        ATTR_ENTITY_ID: self.state.entity_id
                     }, blocking=True, context=data.context)
             else:
-                if self.state.state != cover.STATE_CLOSED:
-                    if params['openPercent'] < 100:
-                        await self.hass.services.async_call(
-                            cover.DOMAIN, cover.SERVICE_CLOSE_COVER, {
-                                ATTR_ENTITY_ID: self.state.entity_id
-                            }, blocking=True, context=data.context)
+                if position is not None:
+                    await self.hass.services.async_call(
+                        cover.DOMAIN, cover.SERVICE_SET_COVER_POSITION, {
+                            ATTR_ENTITY_ID: self.state.entity_id,
+                            cover.ATTR_POSITION: params['openPercent']
+                        }, blocking=True, context=data.context)
                 else:
-                    if params['openPercent'] > 0:
-                        await self.hass.services.async_call(
-                            cover.DOMAIN, cover.SERVICE_OPEN_COVER, {
-                                ATTR_ENTITY_ID: self.state.entity_id
-                            }, blocking=True, context=data.context)
+                    if self.state.state != cover.STATE_CLOSED:
```

Comment:
same

Reply:
Also, we should remove this logic. This makes no sense. We get explicit commands from Google, not toggle commands. We should not have the current state influence how we handle it.